extension/js/common/core/attachment.ts

```diff
@@ -118,6 +118,9 @@ export class Attachment {
       return 'hidden'; // mail.ch does this - although it looks like encrypted msg, it will just contain PGP version eg "Version: 1"
     } else if (Attachment.encryptedMsgNames.includes(this.name)) {
       return 'encryptedMsg';
+    } else if (this.name === 'message' && (this.type === 'application/pgp-encrypted')) {
```

Comment:
I only added `type` check here (Tom mentioned it would be better to check empty body too) because we have to change `Attachment` a little(Need to accept emailBody or isEmailBodyEmpty and need to pass it in several places)

Maybe I think we can reconsider this if user complains about `message` attachment message doesn't get decrypted.

Reply:
This type `application/pgp-encrypted` is used only to contain `Version: 1` as specified here
https://www.rfc-editor.org/rfc/rfc3156#section-4
The actual encrypted message is contained in the second (indexed 1) part of type `Content-Type: application/octet-stream`.

We have a number of such messages in our tests, e.g. `166147ea9bb6669d`
We mostly have (file)name="encrypted.asc" there.

And we also have at least 2 live tests `[ci.test] Microsoft outlook OpenPGP email not recognized (FMfcgzGllVqqBbjHQQRDsSwcZBlMRzDr).eml`
and `[ci.test] Thunderbird html signed + encrypted (FMfcgzGkbDZKPLBqWFzbgWqCrplTQdNz).eml`

The RFC doesn't require a specific filename to be present, so it may as well be "message".
So we ought to create a correct PGP/MIME message with (file)name="message" and see how it passes the mock and live tests.
`GmailElementReplacer` is a bit more complicated thing. It first reacts to `webmailNamePattern` regex to find "suspicious" attachments (and "message" is still present there) -- I think we planned to refactor this out as this regex will tend to grow as various PGP/MIME encrypting clients may introduce their own filename patterns.

And the last thing to know: Sometimes a PGP/MIME message gets reformatted by a mail agent (e.g. `gmail`), so the enclosing type `multipart/encrypted; protocol="application/pgp-encrypted"` simply becomes `multipart/mixed` or something. In this case we're unable to determine whether `application/octet-stream` is a part of PGP/MIME-encrypted or not. That `flowcrypt.compatibilty` message Tom provided looks like a reformatted message. So these tricks with empty body actually make sense.

Reply:
Got it. Thanks for your brief explanation.
Pushed fix. Please review

Reply:
> So we ought to create a correct PGP/MIME message with (file)name="message" and see how it passes the mock and live tests.

Please create tests on such a message